openingd/dualopend.c

```diff
@@ -2727,8 +2727,10 @@ static u8 *opener_commits(struct state *state,
 				      type_to_string(tmpctx,
 						     struct wally_psbt,
 						     tx_state->psbt));
+		tal_free(wscript);
 		return NULL;
 	}
+	tal_free(wscript);
 
```

Comment:
This commit is just weird... the only danger with tmpctx is when you've got an inner io_loop, which is already an anti-pattern, and there's not one here that I can see?